crypto/src/proofs/transparent.rs

```diff
@@ -439,23 +443,18 @@ impl SwapProof {
             return Err(Error::TransmissionKeyMismatch);
         }
 
+        // TODO: do we need to check the swap NFT Asset ID is properly constructed here?
+
```

Comment:
Yeah, we'll need to include (most?) of the fields of the `SwapPlaintext` in the `SwapClaimProof` (and decide exactly which ones), so that we can have the proof check that the asset ID in the note that's being spent is actually a commitment to the data that's used to form the output values.